### PR TITLE
Support configure System log table`s ttl in config.xml

### PR DESCRIPTION
**Pull request by spongedu**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Allow specifying [TTL](https://clickhouse.tech/docs/en/engines/table-engines/mergetree-family/mergetree/#mergetree-table-ttl) to remove old entries from [system log tables](https://clickhouse.tech/docs/en/operations/system-tables/),  using the `<ttl>` attribute in `config.xml`.

### CODE REVIEW

File: programs/server/config.xml [hunk 1]
```diff
@@ -589,6 +589,9 @@
                 toStartOfHour(event_time)
         -->
         <partition_by>toYYYYMM(event_date)</partition_by>
+        <!-- How many days data would be kept. It works only when this option is opened and is set greater than 0.
+        -->
+        <ttl>30</ttl>
```
Comment (akuzm) [inline programs/server/config.xml:594]: Let's name it `<delete_after_days>` or something, just `<ttl>` is too generic. Maybe we'll want to use it to accept generic TTL specification (or you can do it now if you wish, just paste it into table description).

File: src/Interpreters/SystemLog.cpp [hunk 2]
```diff
@@ -65,6 +65,9 @@ std::shared_ptr<TSystemLog> createSystemLog(
         engine = "ENGINE = MergeTree";
         if (!partition_by.empty())
             engine += " PARTITION BY (" + partition_by + ")";
+        int ttl = config.getInt(config_prefix + ".ttl", 0);
```
Comment (akuzm) [inline src/Interpreters/SystemLog.cpp:68]: Please also add a check to the above `if`, that if engine is specified manually, ttl must not be specified and must instead go into the engine specification. Like for `<partition_by>`.

File: programs/server/config.xml (hunk 1 above)
Comment (spongedu) [reply programs/server/config.xml:594]: ok, I think we can keep it as `<ttl>` and make it accept generic `TTL` specification.

Comment (spongedu) [reply programs/server/config.xml:594]: > Let's name it `<delete_after_days>` or something, just `<ttl>` is too generic. Maybe we'll want to use it to accept generic TTL specification (or you can do it now if you wish, just paste it into table description).

ok, I think we can keep it as <ttl> and make it accept generic TTL specification.

File: src/Interpreters/SystemLog.cpp (hunk 2 above)
Comment (spongedu) [reply src/Interpreters/SystemLog.cpp:68]: ok. Will fix here.

**Comment by spongedu** [comment]
@akuzm Thanks for your review. I've made some updates according to our discussion. PTAL.

**Comment by akuzm** [comment]
The test failures are not related.

This feature is very simple, but tricky to test, so I think we don't have to add a test for it.

**Pull request merged**